src/newcharacter.cpp

```diff
@@ -1561,6 +1592,7 @@ tab_direction set_profession( const catacurses::window &w, avatar &u, points_lef
             // Add traits for the new profession (and perhaps scenario, if, for example,
             // both the scenario and old profession require the same trait)
             u.add_traits( points );
+            const int netPointCost = sorted_profs[cur_id]->point_cost() - u.prof->point_cost();
```

Comment:
I guess issue #39208 is from here - `netPointCost` should contain difference between current and new profession, but character profession was already set to selected profession just 4 lines above, so it always returns 0.